assets/src/edit-story/components/panels/design/textStyle/fontPicker.js

```diff
@@ -113,17 +113,14 @@ const FontPicker = forwardRef(function FontPicker(
     ]
   );
 
-  const fontMap = useMemo(
-    () =>
-      [...fonts, ...recentFonts, ...curatedFonts].reduce(
-        (lookup, option) => ({
-          ...lookup,
-          [option.id]: option,
-        }),
-        {}
-      ),
-    [fonts, recentFonts, curatedFonts]
-  );
+  const fontMap = useMemo(() => {
+    const map = new Map();
+    // curatedFonts and recentFonts are subsets of fonts.
+    fonts.forEach((f) => {
+      map.set(f.id, f);
+    });
+    return map;
+  }, [fonts]);
 
   const onObserve = (observedFonts) => {
     ensureMenuFontsLoaded(
```

Comment:
@choumx The filtering below (`fontMap[fontName]?.service` part) should also be updated to use the new format of the `fontMap`, otherwise the font preview won't work anymore.

Reply:
Good catch, thanks!